aten/src/ATen/native/mps/operations/LinearAlgebra.mm

```diff
@@ -243,62 +238,62 @@ Tensor addr_mps(const Tensor& self, const Tensor& vec1, const Tensor& vec2, cons
           MPSGraph* mpsGraph = mps::make_mps_graph();
           newCachedGraph = new CachedGraph(mpsGraph);
 
-          MPSGraphTensor* t1 = mps::mpsGraphRankedPlaceHolder(mpsGraph, getMPSDataType(vec1), inputShape);
-          MPSGraphTensor* t2 = mps::mpsGraphRankedPlaceHolder(mpsGraph, getMPSDataType(vec2), otherShape);
-          MPSGraphTensor* selfTensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, *self_);
-
-          // Intermediate as placeholder
-          MPSGraphTensor* productTensor = [mpsGraph matrixMultiplicationWithPrimaryTensor:t1
-                                                                          secondaryTensor:t2
-                                                                                     name:@"MM/(vec1Xvec2)"];
+          MPSGraphTensor* inputTensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, input);
+          MPSGraphTensor* batch1Tensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, batch1);
+          MPSGraphTensor* batch2Tensor = mps::mpsGraphRankedPlaceHolder(mpsGraph, batch2);
 
           // Intermediates for beta and alpha
           MPSGraphTensor* betaTensor = [mpsGraph constantWithScalar:beta.toDouble()
-                                                           dataType:getMPSScalarType((*self_).scalar_type())];
+                                                           dataType:getMPSScalarType(input.scalar_type())];
           MPSGraphTensor* alphaTensor = [mpsGraph constantWithScalar:alpha.toDouble()
-                                                            dataType:getMPSScalarType(vec1.scalar_type())];
+                                                            dataType:getMPSScalarType(batch1.scalar_type())];
 
-          // Intermediates for multiplying by beta and alpha
-          MPSGraphTensor* productTimesAlphaTensor = [mpsGraph multiplicationWithPrimaryTensor:productTensor
-                                                                              secondaryTensor:alphaTensor
-                                                                                         name:@"MM/alpha*(vec1Xvec2)"];
-          MPSGraphTensor* selfTimesBetaTensor = selfTensor;
-          if (is_beta_non_zero) {
-            selfTimesBetaTensor = [mpsGraph multiplicationWithPrimaryTensor:selfTensor
-                                                            secondaryTensor:betaTensor
-                                                                       name:@"MM/beta*input"];
```

Comment:
This is changing quite a few things.  Can we make separate PRs where it's doing cleanup? vs where its refactoring code ?

Reply:
Replied on the Slack.